From 0da36b3abe69cbdfbc80fc39a9281c2190232dc9 Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Tue, 4 Jun 2019 13:14:51 +0200
Subject: [PATCH 282/386] arm64: dts: marvell: remove references to PCIe IO
 space.

Hitherto the PCIe IO spaces were pointing to not-mapped or not valid
addresses.  E.g. in case of a80x0, the PCIe IO 0xf900000 in fact is
mapped to SPI1-CP1 but the PCIe IO description points to this address.
Therefore remove all PCIe IO space related entries.

Change-Id: I7d71fe5e3bf5e8fbe62465d4a7f6ae9e7cdacc22
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: https://sj1git1.cavium.com/10473
Tested-by: Stefan Chulski <Stefan.Chulski@cavium.com>
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 .../arm64/boot/dts/marvell/armada-3900-db.dtsi |  6 ++----
 arch/arm64/boot/dts/marvell/armada-70x0.dtsi   |  4 ----
 arch/arm64/boot/dts/marvell/armada-80x0.dtsi   |  8 --------
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi  | 18 +++---------------
 arch/arm64/boot/dts/marvell/cn9130.dtsi        |  2 --
 arch/arm64/boot/dts/marvell/cn9131-db.dtsi     |  4 ----
 arch/arm64/boot/dts/marvell/cn9132-db.dtsi     |  4 ----
 7 files changed, 5 insertions(+), 41 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-3900-db.dtsi b/arch/arm64/boot/dts/marvell/armada-3900-db.dtsi
index dde5993de18e..1ef0539cf489 100644
--- a/arch/arm64/boot/dts/marvell/armada-3900-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-3900-db.dtsi
@@ -141,15 +141,13 @@
 };
 
 &cp0_pcie0 {
-	ranges = <0x81000000 0 0xf7000000 0  0xf7000000 0 0x10000
-		  0x82000000 0 0xc0000000 0  0xc0000000 0 0x20000000>;
+	ranges = <0x82000000 0 0xc0000000 0  0xc0000000 0 0x20000000>;
 	phys = <&cp0_comphy0 0>;
 	phy-names = "pcie-phy0";
 };
 
 &cp0_pcie2 {
-	ranges = <0x81000000 0 0xf7020000 0  0xf7020000 0 0x10000
-		  0x82000000 0 0xf8000000 0  0xf8000000 0 0xf00000>;
+	ranges = <0x82000000 0 0xf8000000 0  0xf8000000 0 0xf00000>;
 	phys = <&cp0_comphy5 2>;
 	phy-names = "pcie-phy2";
 };
diff --git a/arch/arm64/boot/dts/marvell/armada-70x0.dtsi b/arch/arm64/boot/dts/marvell/armada-70x0.dtsi
index cca2a6a4cfe1..736826d68839 100644
--- a/arch/arm64/boot/dts/marvell/armada-70x0.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-70x0.dtsi
@@ -11,9 +11,7 @@
 #define CP110_NUM				0
 #define CP110_BASE				0xf2000000
 #define CP110_PCIE_MEM_SIZE(iface)		0xf00000
-#define CP110_PCIEx_CPU_IO_BASE(iface)		(0xf9000000 + (iface) *  0x10000)
 #define CP110_PCIEx_CPU_MEM_BASE(iface)		(0xf6000000 + (iface) *  0x1000000)
-#define CP110_PCIEx_BUS_IO_BASE(iface)		(CP110_PCIEx_CPU_IO_BASE(iface))
 #define CP110_PCIEx_BUS_MEM_BASE(iface)		(CP110_PCIEx_CPU_MEM_BASE(iface))
 #define CP110_PCIE_BUS_MEM_CFG			(0x82000000)
 
@@ -22,9 +20,7 @@
 #undef CP110_NUM
 #undef CP110_BASE
 #undef CP110_PCIE_MEM_SIZE
-#undef CP110_PCIEx_CPU_IO_BASE
 #undef CP110_PCIEx_CPU_MEM_BASE
-#undef CP110_PCIEx_BUS_IO_BASE
 #undef CP110_PCIEx_BUS_MEM_BASE
 #undef CP110_PCIE_BUS_MEM_CFG
 
diff --git a/arch/arm64/boot/dts/marvell/armada-80x0.dtsi b/arch/arm64/boot/dts/marvell/armada-80x0.dtsi
index 575e324f2648..1941e9a79f16 100644
--- a/arch/arm64/boot/dts/marvell/armada-80x0.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-80x0.dtsi
@@ -11,9 +11,7 @@
 #define CP110_NUM				0
 #define CP110_BASE				0xf2000000
 #define CP110_PCIE_MEM_SIZE(iface)		0xf00000
-#define CP110_PCIEx_CPU_IO_BASE(iface)		(0xf9000000 + (iface) *  0x10000)
 #define CP110_PCIEx_CPU_MEM_BASE(iface)		(0xf6000000 + (iface) *  0x1000000)
-#define CP110_PCIEx_BUS_IO_BASE(iface)		(CP110_PCIEx_CPU_IO_BASE(iface))
 #define CP110_PCIEx_BUS_MEM_BASE(iface)		(CP110_PCIEx_CPU_MEM_BASE(iface))
 #define CP110_PCIE_BUS_MEM_CFG			(0x82000000)
 
@@ -22,9 +20,7 @@
 #undef CP110_NUM
 #undef CP110_BASE
 #undef CP110_PCIE_MEM_SIZE
-#undef CP110_PCIEx_CPU_IO_BASE
 #undef CP110_PCIEx_CPU_MEM_BASE
-#undef CP110_PCIEx_BUS_IO_BASE
 #undef CP110_PCIEx_BUS_MEM_BASE
 #undef CP110_PCIE_BUS_MEM_CFG
 
@@ -34,9 +30,7 @@
 #define CP110_NUM				1
 #define CP110_BASE				0xf4000000
 #define CP110_PCIE_MEM_SIZE(iface)		0xf00000
-#define CP110_PCIEx_CPU_IO_BASE(iface)		(0xfd000000 + (iface) *  0x10000)
 #define CP110_PCIEx_CPU_MEM_BASE(iface)		(0xfa000000 + (iface) *  0x1000000)
-#define CP110_PCIEx_BUS_IO_BASE(iface)		(CP110_PCIEx_CPU_IO_BASE(iface))
 #define CP110_PCIEx_BUS_MEM_BASE(iface)		(CP110_PCIEx_CPU_MEM_BASE(iface))
 #define CP110_PCIE_BUS_MEM_CFG			(0x82000000)
 
@@ -46,9 +40,7 @@
 #undef CP110_NUM
 #undef CP110_BASE
 #undef CP110_PCIE_MEM_SIZE
-#undef CP110_PCIEx_CPU_IO_BASE
 #undef CP110_PCIEx_CPU_MEM_BASE
-#undef CP110_PCIEx_BUS_IO_BASE
 #undef CP110_PCIEx_BUS_MEM_BASE
 #undef CP110_PCIE_BUS_MEM_CFG
 
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index b25bedfd81b5..b9a37b6e9eb4 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -568,12 +568,8 @@
 		msi-parent = <&gic_v2m0>;
 		bus-range = <0 0xff>;
 		ranges =
-			/* downstream I/O */
-			<0x81000000 U64_TO_U32_H(CP110_PCIEx_BUS_IO_BASE(0))
-			U64_TO_U32_L(CP110_PCIEx_BUS_IO_BASE(0)) U64_TO_U32_H(CP110_PCIEx_CPU_IO_BASE(0))
-			U64_TO_U32_L(CP110_PCIEx_CPU_IO_BASE(0)) 0 0x10000
 			/* non-prefetchable memory */
-			CP110_PCIE_BUS_MEM_CFG U64_TO_U32_H(CP110_PCIEx_BUS_MEM_BASE(0))
+			<CP110_PCIE_BUS_MEM_CFG U64_TO_U32_H(CP110_PCIEx_BUS_MEM_BASE(0))
 			U64_TO_U32_L(CP110_PCIEx_BUS_MEM_BASE(0)) U64_TO_U32_H(CP110_PCIEx_CPU_MEM_BASE(0))
 			U64_TO_U32_L(CP110_PCIEx_CPU_MEM_BASE(0))
 			U64_TO_U32_H(CP110_PCIE_MEM_SIZE(0)) U64_TO_U32_L(CP110_PCIE_MEM_SIZE(0))>;
@@ -601,12 +597,8 @@
 		msi-parent = <&gic_v2m0>;
 		bus-range = <0 0xff>;
 		ranges =
-			/* downstream I/O */
-			<0x81000000 U64_TO_U32_H(CP110_PCIEx_BUS_IO_BASE(1))
-			U64_TO_U32_L(CP110_PCIEx_BUS_IO_BASE(1)) U64_TO_U32_H(CP110_PCIEx_CPU_IO_BASE(1))
-			U64_TO_U32_L(CP110_PCIEx_CPU_IO_BASE(1)) 0 0x10000
 			/* non-prefetchable memory */
-			CP110_PCIE_BUS_MEM_CFG U64_TO_U32_H(CP110_PCIEx_BUS_MEM_BASE(1))
+			<CP110_PCIE_BUS_MEM_CFG U64_TO_U32_H(CP110_PCIEx_BUS_MEM_BASE(1))
 			U64_TO_U32_L(CP110_PCIEx_BUS_MEM_BASE(1)) U64_TO_U32_H(CP110_PCIEx_CPU_MEM_BASE(1))
 			U64_TO_U32_L(CP110_PCIEx_CPU_MEM_BASE(1))
 			U64_TO_U32_H(CP110_PCIE_MEM_SIZE(1)) U64_TO_U32_L(CP110_PCIE_MEM_SIZE(1))>;
@@ -634,12 +626,8 @@
 		msi-parent = <&gic_v2m0>;
 		bus-range = <0 0xff>;
 		ranges =
-			/* downstream I/O */
-			<0x81000000 U64_TO_U32_H(CP110_PCIEx_BUS_IO_BASE(2))
-			U64_TO_U32_L(CP110_PCIEx_BUS_IO_BASE(2)) U64_TO_U32_H(CP110_PCIEx_CPU_IO_BASE(2))
-			U64_TO_U32_L(CP110_PCIEx_CPU_IO_BASE(2)) 0 0x10000
 			/* non-prefetchable memory */
-			CP110_PCIE_BUS_MEM_CFG U64_TO_U32_H(CP110_PCIEx_BUS_MEM_BASE(2))
+			<CP110_PCIE_BUS_MEM_CFG U64_TO_U32_H(CP110_PCIEx_BUS_MEM_BASE(2))
 			U64_TO_U32_L(CP110_PCIEx_BUS_MEM_BASE(2)) U64_TO_U32_H(CP110_PCIEx_CPU_MEM_BASE(2))
 			U64_TO_U32_L(CP110_PCIEx_CPU_MEM_BASE(2))
 			U64_TO_U32_H(CP110_PCIE_MEM_SIZE(2)) U64_TO_U32_L(CP110_PCIE_MEM_SIZE(2))>;
diff --git a/arch/arm64/boot/dts/marvell/cn9130.dtsi b/arch/arm64/boot/dts/marvell/cn9130.dtsi
index bbcbd04ca6a6..7ddba373d8be 100644
--- a/arch/arm64/boot/dts/marvell/cn9130.dtsi
+++ b/arch/arm64/boot/dts/marvell/cn9130.dtsi
@@ -32,10 +32,8 @@
 
 /* CP110-0 Settings */
 #define CP110_NUM			0
-#define CP110_PCIEx_CPU_IO_BASE(iface)	(0xf8000000 + (iface) * 0x10000)
 #define CP110_PCIEx_CPU_MEM_BASE(iface)	((iface == 0) ? 0xc0000000 : \
 					 (0xe0000000 + (iface - 1) * 0x1000000))
-#define CP110_PCIEx_BUS_IO_BASE(iface)	(CP110_PCIEx_CPU_IO_BASE(iface))
 #define CP110_PCIEx_BUS_MEM_BASE(iface)	(CP110_PCIEx_CPU_MEM_BASE(iface))
 
 #include "armada-cp110.dtsi"
diff --git a/arch/arm64/boot/dts/marvell/cn9131-db.dtsi b/arch/arm64/boot/dts/marvell/cn9131-db.dtsi
index ee18a493e622..a752f17f46a8 100644
--- a/arch/arm64/boot/dts/marvell/cn9131-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/cn9131-db.dtsi
@@ -7,17 +7,13 @@
 
 #undef CP110_NUM
 #undef CP110_PCIE_MEM_SIZE
-#undef CP110_PCIEx_CPU_IO_BASE
 #undef CP110_PCIEx_CPU_MEM_BASE
-#undef CP110_PCIEx_BUS_IO_BASE
 #undef CP110_PCIEx_BUS_MEM_BASE
 
 /* CP110-1 Settings */
 #define CP110_NUM			1
 #define CP110_PCIE_MEM_SIZE(iface)	(0xf00000)
-#define CP110_PCIEx_CPU_IO_BASE(iface)	(0xf8030000 + (iface) * 0x10000)
 #define CP110_PCIEx_CPU_MEM_BASE(iface)	(0xe2000000 + (iface) * 0x1000000)
-#define CP110_PCIEx_BUS_IO_BASE(iface)	(CP110_PCIEx_CPU_IO_BASE(iface))
 #define CP110_PCIEx_BUS_MEM_BASE(iface)	(CP110_PCIEx_CPU_MEM_BASE(iface))
 
 #include "armada-cp110.dtsi"
diff --git a/arch/arm64/boot/dts/marvell/cn9132-db.dtsi b/arch/arm64/boot/dts/marvell/cn9132-db.dtsi
index 6e1aa747a9a5..b0a7e71f10e5 100644
--- a/arch/arm64/boot/dts/marvell/cn9132-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/cn9132-db.dtsi
@@ -7,17 +7,13 @@
 
 #undef CP110_NUM
 #undef CP110_PCIE_MEM_SIZE
-#undef CP110_PCIEx_CPU_IO_BASE
 #undef CP110_PCIEx_CPU_MEM_BASE
-#undef CP110_PCIEx_BUS_IO_BASE
 #undef CP110_PCIEx_BUS_MEM_BASE
 
 /* CP110-1 Settings */
 #define CP110_NUM			2
 #define CP110_PCIE_MEM_SIZE(iface)	(0xf00000)
-#define CP110_PCIEx_CPU_IO_BASE(iface)	(0xf8060000 + (iface) * 0x10000)
 #define CP110_PCIEx_CPU_MEM_BASE(iface)	(0xe5000000 + (iface) * 0x1000000)
-#define CP110_PCIEx_BUS_IO_BASE(iface)	(CP110_PCIEx_CPU_IO_BASE(iface))
 #define CP110_PCIEx_BUS_MEM_BASE(iface)	(CP110_PCIEx_CPU_MEM_BASE(iface))
 
 #include "armada-cp110.dtsi"
-- 
2.17.1

